core/src/main/java/org/lflang/generator/docker/CDockerGenerator.java

```diff
@@ -30,79 +32,47 @@ public String defaultImage() {
-        "COPY . src-gen",
-        compileCommand,
-        "",
-        "FROM " + baseImage,
-        "WORKDIR /lingua-franca",
-        "RUN mkdir bin",
-        "COPY --from=builder /lingua-franca/"
-            + lfModuleName
-            + "/bin/"
-            + lfModuleName
-            + " ./bin/"
-            + lfModuleName,
-        "",
-        "# Use ENTRYPOINT not CMD so that command-line arguments go through",
-        "ENTRYPOINT [\"./bin/" + lfModuleName + "\"]",
-        "");
+  public List<String> defaultEntryPoint() {
+    return List.of("./bin/" + context.getFileConfig().name);
   }
 
   @Override
-  protected String generateRunForBuildDependencies() {
+  protected String generateRunForInstallingDeps() {
     var config = context.getTargetConfig();
     var compiler = config.target == Target.CCPP ? "g++" : "gcc";
-    if (baseImage().equals(defaultImage())) {
-      return """
-             # Install build dependencies
-             RUN set -ex && apk add --no-cache %s musl-dev cmake make
-             # Optional user specified run command
-             %s
-             """
-          .formatted(compiler, userRunCommand());
+    if (builderBase().equals(defaultImage())) {
+      return "RUN set -ex && apk add --no-cache %s musl-dev cmake make".formatted(compiler);
     } else {
-      return """
-             # Optional user specified run command
-             %s
-             # Check for build dependencies
-             RUN which make && which cmake && which %s
-             """
-          .formatted(userRunCommand(), compiler);
+      return "# (Skipping installation of build dependencies; custom base image.)";
     }
   }
 
-  /** Return the default compile command for the C docker container. */
-  protected String generateCompileCommand() {
-    var ccompile =
-        new CCompiler(
-            context.getTargetConfig(),
-            context.getFileConfig(),
-            context.getErrorReporter(),
-            context.getTargetConfig().target == Target.C);
-    return String.join(
-        "\n",
-        "RUN set -ex && \\",
-        "mkdir bin && \\",
-        String.format(
-            "%s -DCMAKE_INSTALL_BINDIR=./bin -S src-gen -B bin && \\",
-            ccompile.compileCmakeCommand()),
-        "cd bin && \\",
-        "make all");
+  @Override
+  protected List<String> defaultBuildCommands() {
+    try {
+      var ccompile =
+          new CCompiler(
+              context.getTargetConfig(),
+              new CFileConfig(
+                  context.getFileConfig().resource,
+                  Path.of("/lingua-franca", context.getFileConfig().name),
+                  false),
+              context.getErrorReporter(),
+              context.getTargetConfig().target == Target.CCPP);
+      return List.of(
+          "mkdir -p bin",
+          String.format(
+              "%s -DCMAKE_INSTALL_BINDIR=./bin -S src-gen -B bin", ccompile.compileCmakeCommand()),
+          "cd bin",
+          "make all",
```

Comment:
The usual name for this directory is `build`, not `bin`, and people usually keep their `build` and `bin` directories separate so that the `bin` directory only contains executables. Is there a particular reason why it was done in this way?

Reply:
Not that I know of. It says `BINDIR` though. Doesn't that suggest `bin`?

Reply:
Sorry, I meant where it says `-B bin` and `mkdir -p bin`. The `BINDIR` probably should be `bin`, but what I thought was odd was that it was the same as build build directory that was created with `mkdir -p` and passed to `cmake` using the `-B` flag.

Reply:
IDK, maybe @cmnrd could weigh in on this?

Reply:
Don't know either. Choosing `bin` for the build directory name is indeed strange. In particular, since we also install in `bin` and thus the separation between build and bin directory seems to be gone.

Reply:
:man_shrugging: I don't see anything actionable here unless there is a concrete alternative proposed.

Reply:
The alternative is using:
```
"%s -DCMAKE_INSTALL_BINDIR=./bin -S src-gen -B build", ...
cd build,
```
which seems more sane.

Reply:
@petervdonovan would you want to address this? If so, please push a fix and hit resolve. If no, please just resolve this conversation to unblock the merge.